allensdk/brain_observatory/filtered_running_speed/stim_file.py

```diff
@@ -0,0 +1,182 @@
+
+    @property
+    def angular_wheel_velocity(self):
+        """ Extract the mean angular velocity of the running wheel
+        (degrees / s) for each frame.
+        """
+        return self.frames_per_second * self.angular_wheel_rotation
+
+    @property
+    def angular_wheel_rotation(self):
+        """ Extract the total rotation of the running wheel on each frame.
+        """
+        return self._extract_running_array("dx")
+
+    @property
+    def vsig(self):
+        """Running speed signal voltage
+        """
+        return self._extract_running_array("vsig")
+
+    @property
+    def vin(self):
+        return self._extract_running_array("vin")
+
+    def __init__(self, data, **kwargs):
+        self._data = data
+
+    def _extract_running_array(self, key):
+        try:
+            result = self._data["items"]["foraging"]["encoders"][0][key]
+        except (KeyError, IndexError):
+            try:
+                result = self._data[key]
+            except KeyError:
+                raise KeyError(
+                    f"Unable to extract {key} from this stimulus pickle!"
+                )
+
+        return np.array(result)
+
+    @classmethod
+    def factory(cls, path, **kwargs):
+        data = pd.read_pickle(path)
+        return cls(data, **kwargs)
+
+
+class BehaviorPickleFile(object):
+    """A helper class to abstract common data/metadata lookup operations
+    with a visual behavior session pkl file.
+    """
+
+    def _behavior_params(self) -> dict:
+        """Shortcut to access behavior pkl behavior specific params"""
+        return self._data["items"]["behavior"]
+
+    @property
+    def data(self) -> dict:
+        """Pkl data in dictionary form"""
+        return self._data
+
+    @property
+    def presentation_intervals(self) -> np.ndarray:
+        """Get an array of interval times (in ms) between stimulus
+        presentations
+        """
+        behavior = self._behavior_params()
+        return np.array(behavior["intervalsms"])
+
+    @property
+    def image_set(self) -> str:
+        """Get the name of the image set that was presented during the pkl
+        session.
+
+        Names will look something like:
+        'Natural_Images_Lum_Matched_set_ophys_G_2019'
+        """
+        behavior = self._behavior_params()
+        image_set_path = behavior["params"]["stimulus"]["params"]["image_set"]
+        image_set = image_set_path.split('/')[-1].split('.')[0]
+        return image_set
+
+    @property
+    def num_frames(self) -> int:
+        """Get the number of stimulus frames presented during pkl session."""
+        return len(self.presentation_intervals) + 1
+
+    @property
+    def reward_frames(self) -> np.ndarray:
+        """Get the frames where a reward was delivered during pkl session."""
+        behavior = self._behavior_params()
+        reward_frames = behavior["rewards"][0]["reward_times"][:, 1]
+        return reward_frames
+
+    def __init__(self, data, **kwargs):
```

Comment:
Please move this to the top of the class. **kwargs are not used. please remove it

Reply:
I've removed the dependency on the stim file library.